source/adapters/opencl/context.hpp

```diff
@@ -10,9 +10,74 @@
 #pragma once
 
 #include "common.hpp"
+#include "device.hpp"
 
-namespace cl_adapter {
-ur_result_t
-getDevicesFromContext(ur_context_handle_t hContext,
-                      std::unique_ptr<std::vector<cl_device_id>> &DevicesInCtx);
-}
+#include <vector>
+
+struct ur_context_handle_t_ {
+  using native_type = cl_context;
+  native_type Context;
+  std::vector<ur_device_handle_t> Devices;
+  uint32_t DeviceCount;
+  std::atomic<uint32_t> RefCount = 0;
+
+  ur_context_handle_t_(native_type Ctx, uint32_t DevCount,
+                       const ur_device_handle_t *phDevices)
+      : Context(Ctx), DeviceCount(DevCount) {
+    for (uint32_t i = 0; i < DeviceCount; i++) {
+      Devices.emplace_back(phDevices[i]);
+      urDeviceRetain(phDevices[i]);
+    }
+    urContextRetain(this);
```

Comment:
I think we could avoid the function call here and just initialize `RefCount` to 1 in the initializer list, like what the cuda adapter does https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/cuda/context.hpp#L82

Reply:
Fixed that